TicTacToe/test/tictactoe/TicTacToeTest.java

```diff
@@ -3,34 +3,142 @@
+     */
     @Test
-    public void testIsFull() {
+    public void testIsFull()
+    {
+        ttt = new TicTacToe();
+        assertFalse(ttt.isFull());
+        for (int r = 0; r < 3; r++) //Fills the board
+        {
+            for (int c = 0; c < 3; c++)
+            {
+                ttt.playMove('X', r, c);
+            }
+        }
+        assertTrue(ttt.isFull());
     }
 
+    /**
+     * Test of IsCat method, of class TicTacToe.
+     */
     @Test
-    public void testIsCat() {
+    public void testIsCat()
+    {
+        ttt = new TicTacToe();
+        assertFalse(ttt.isCat());
+        for (int r = 0; r < 3; r++) //Full with winner
+        {
+            for (int c = 0; c < 3; c++)
+            {
+                ttt.playMove('X', r, c);
+            }
+        }
+        assertFalse(ttt.isCat());
+        ttt = new TicTacToe();
+        ttt.playMove('X', 0, 0); //Full without winner
+        ttt.playMove('O', 0, 1);
+        ttt.playMove('X', 0, 2);
+        ttt.playMove('X', 1, 0);
+        ttt.playMove('O', 1, 1);
+        ttt.playMove('X', 1, 2);
+        ttt.playMove('O', 2, 0);
+        ttt.playMove('X', 2, 1);
+        ttt.playMove('O', 2, 2);
+        assertTrue(ttt.isCat());        
     }
 
+    /**
+     * Test of IsValid method, of class TicTacToe.
+     */
     @Test
-    public void testIsValid() {
+    public void testIsValid()
+    {
+        ttt = new TicTacToe();
+        assertTrue(ttt.isValid(0, 0)); //All valid spaces
+        assertTrue(ttt.isValid(0, 1));
+        assertTrue(ttt.isValid(0, 2));
+        assertTrue(ttt.isValid(1, 0));
+        assertTrue(ttt.isValid(1, 1));
+        assertTrue(ttt.isValid(1, 2));
+        assertTrue(ttt.isValid(2, 0));
+        assertTrue(ttt.isValid(2, 1));
+        assertTrue(ttt.isValid(2, 2));
+        
+        assertFalse(ttt.isValid(3, 3)); //Some invalid spaces
+        assertFalse(ttt.isValid(3, -1));
+        assertFalse(ttt.isValid(-1, 3));
+        assertFalse(ttt.isValid(-1, -1));
+        
     }
 
+    /**
+     * Test of PlayerAt method, of class TicTacToe.
+     */
     @Test
-    public void testPlayerAt() {
+    public void testPlayerAt()
+    {
+        ttt = new TicTacToe();
+        ttt.playMove('X', 0, 0); //Set all squares and check them
+        ttt.playMove('O', 0, 1);
+        ttt.playMove('X', 0, 2);
+        ttt.playMove('X', 1, 0);
+        ttt.playMove('O', 1, 1);
+        ttt.playMove('X', 1, 2);
+        ttt.playMove('O', 2, 0);
+        ttt.playMove('X', 2, 1);
+        ttt.playMove('O', 2, 2);
+        assertEquals(ttt.playerAt(0, 0), 'X');
+        assertEquals(ttt.playerAt(0, 1), 'O');
+        assertEquals(ttt.playerAt(0, 2), 'X');
+        assertEquals(ttt.playerAt(1, 0), 'X');
+        assertEquals(ttt.playerAt(1, 1), 'O');
+        assertEquals(ttt.playerAt(1, 2), 'X');
+        assertEquals(ttt.playerAt(2, 0), 'O');
+        assertEquals(ttt.playerAt(2, 1), 'X');
+        assertEquals(ttt.playerAt(2, 2), 'O');
```

Comment:
Should also cover case where no player is present.  Should also check an invalid space.